File: ED62/a.rb
require 'set'
#if File.exists?("input")
#  $stdin = File.open("input")
#  $stdout = File.open("output","w")
#end
n = gets.to_i
arr = gets.chomp.split(" ").map(&:to_i)

i=0
ans=0
mySet = Set.new

while(i<n)
	ans+=1;
    loop do
		mySet.add(arr[i])
    	if(mySet.include?(i+1))
    		mySet.delete(i+1)
    	end
    	i+=1
    	break if mySet.empty?
    end
end

puts ans

File: ED62/a.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	
//	#ifndef ONLINE_JUDGE
//    	freopen("input", "r", stdin);
//    	freopen("output", "w", stdout);
//	#endif
	ios_base::sync_with_stdio(false);
    cin.tie(NULL);


    set<int> mySet;

    int n;
    cin>>n;

    int arr[n];
    for(int i=0;i<n;i++){
    	cin>>arr[i];
    }

    int i=0;
    int ans=0;
    while(i<n){
    	ans++;
	    mySet.insert(arr[i]);
    	cout<<mySet.size()<<" "<<i<<endl;
    	while(mySet.size()>0){
    		if(mySet.find(i+1)!=mySet.end())
	    		mySet.erase(i+1);
    		i++;
    		mySet.insert(arr[i]);
    	}
    }

    cout<<ans<<endl;
}
